internal/templating/services/templates_pvc.go

```diff
@@ -51,106 +52,215 @@ func GeneratePVCTemplate(
-				pvc.Spec = corev1.PersistentVolumeClaimSpec{
-					AccessModes: []corev1.PersistentVolumeAccessMode{
-						serviceTypeValues.Volumes.PersistentVolumeType,
-					},
-					Resources: corev1.ResourceRequirements{
-						Requests: corev1.ResourceList{
-							"storage": *resource.NewQuantity(volumeSize, resource.BinarySI),
-						},
-					},
+				if pvc != nil {
+					// end PVC template
+					result = append(result, *pvc)
 				}
-				if serviceTypeValues.Volumes.PersistentVolumeType == corev1.ReadWriteMany {
-					pvc.Spec.StorageClassName = helpers.StrPtr("bulk")
+			}
+		}
+	}
+	// check for any additional volumes
+	for _, vol := range buildValues.Volumes {
+		if vol.Create {
+			exists := false
+			// check if an existing pvc that will be created as a default persistent volume hasn't also been defined as an additional volume
+			// this will prevent creating another volume named the same
+			for _, cpvc := range result {
+				if lagoon.GetLagoonVolumeName(cpvc.Name) == vol.Name {
+					exists = true
 				}
-				if buildValues.RWX2RWO || buildValues.IsCI {
-					// this should be a rwo volume in CI and if the rwx2rwo flag is enabled
-					pvc.Spec.AccessModes = []corev1.PersistentVolumeAccessMode{
-						corev1.ReadWriteOnce,
-					}
+			}
+			if !exists {
+				pvc, err := generateAdditionalPVC(buildValues, vol, labels, annotations)
+				if err != nil {
+					return nil, err
 				}
-				// end PVC template
 				result = append(result, *pvc)
 			}
 		}
 	}
 	return result, nil
 }
+
+// generateDefaultPVC default volumes have different labels/annotations to additional values, and also handle some configuration a bit differently
+func generateDefaultPVC(buildValues generator.BuildValues,
+	serviceValues generator.ServiceValues,
+	val servicetypes.ServiceType,
+	labels, annotations map[string]string,
+) (*corev1.PersistentVolumeClaim, error) {
+	if serviceValues.PersistentVolumeName != "" {
+		if serviceValues.PersistentVolumeName != serviceValues.OverrideName {
+			// this service base volume is not needed because it is created by a different service
+			// lagoon legacy templates allowed due to a funny templating issue, for multiple "basic" types to mount one volume
+			// from one main service, across multiple services of the same type
+			return nil, nil
+		}
+	}
+	serviceTypeValues := &servicetypes.ServiceType{}
+	helpers.DeepCopy(val, serviceTypeValues)
+	persistentVolumeSize := serviceTypeValues.Volumes.PersistentVolumeSize
+	if serviceValues.PersistentVolumeSize != "" {
+		persistentVolumeSize = serviceValues.PersistentVolumeSize
+	}
+	serviceType := &servicetypes.ServiceType{}
+	helpers.DeepCopy(val, serviceType)
+
+	additionalLabels := map[string]string{}
+	additionalAnnotations := map[string]string{}
+
+	additionalLabels["app.kubernetes.io/name"] = serviceType.Name
+	additionalLabels["app.kubernetes.io/instance"] = serviceValues.OverrideName
+	additionalLabels["lagoon.sh/template"] = fmt.Sprintf("%s-%s", serviceType.Name, "0.1.0")
```

Comment:
does it make any sense to move the magic string `0.1.0` into a const?

Reply:
yeah, this is used in numerous places, i'll address it across all the templates in a separate pr